src/raqm.c

```diff
@@ -1126,3 +1136,372 @@ _raqm_u32_to_u8_index (raqm_t   *rq,
                                        output);
   return length;
 }
+
+/* Convert index from UTF-8 to UTF-32 */
+static uint32_t
+_raqm_u8_to_u32_index (raqm_t   *rq,
+                       uint32_t  index)
+{
+  FriBidiStrIndex length;
+  uint32_t *output = malloc((sizeof (uint32_t) * index) + 1);
+
+  length = fribidi_charset_to_unicode (FRIBIDI_CHAR_SET_UTF8,
+                                       rq->text_utf8,
+                                       index,
+                                       output);
+  return length;
+}
+
+static bool
+_allowed_grapheme_boundary(hb_codepoint_t lChar,
+                                       hb_codepoint_t rChar);
+
+static bool
+_in_Hangul_Syllable(hb_codepoint_t Char);
+
+/**
+ * raqm_index_to_position:
+ * @rq: a #raqm_t.
+ * @index: character index.
```

Comment:
`index` is used for both input and output, this should be clearly mentioned and explained in the documentation.